### ISSUE TITLE
Remove third party dependencies

### ISSUE THREAD

**Issue by jackmatt2** [origin-energy/java-snapshot-testing#14]
Remove dependencies so we don't need to shadow as many items.

**Comment by ShishkinDmitriy** [comment]
I think current shadowing of some libs has a problem.
For example, I can't extend built-in `JacksonSnapshotSerializer` and add 3rd-party Jackson Module because serializer require **shadow.** com.fasterxml.jackson.databind.Module;

**Comment by jackmatt2** [comment]
Yeah I found this too.  Agree that we should probably require the user to explicitly add their jackson dependency.

**Comment by ShishkinDmitriy** [comment]
Looks like it is needed gradle analogue of maven's `<optional>true</optional>`

**Comment by jackmatt2** [comment]
The reason I did this originally was to avoid the problem of conflicting Jackson dependencies in the classpath between my version and theirs.  By using their version of Jackson we open up the possibility of a ClassNotFoundException or something.  I'm not saying we don't do it but it's worth thinking about this scenario too.

Now that we support custom serialization I see the potential of a separate project `java-snapshot-testing-jackson` which could allow the client to supply their own Jackson version maybe.

There are just pros and cons of both methods.

**Comment by NicklasWallgren** [comment]
I encountered this issue as well, while trying to add https://github.com/monitorjbl/json-view as a `Jackson Module`.

**Comment by jackmatt2** [comment]
Fixed as of release 2.3.0

Note: you need to manually add these dependencies now too

```
    // Required java-snapshot-testing peer dependencies
   testImplementation 'com.fasterxml.jackson.core:jackson-core:2.11.3'
   testImplementation 'com.fasterxml.jackson.core:jackson-databind:2.11.3'
   testImplementation 'com.fasterxml.jackson.datatype:jackson-datatype-jdk8:2.11.3'
   testImplementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.11.3'
```
